src/api/route-services/samples.js

```diff
@@ -92,29 +105,94 @@ class SamplesService {
     }
   }
 
-  async deleteSamples(projectUuid, experimentId) {
-    logger.log(`Deleting sample for project ${projectUuid} and expId ${experimentId}`);
+  async deleteSamplesFromS3(projectUuid, sampleUuids) {
+    const s3 = new AWS.S3();
+
+    const sampleObjectsToDelete = sampleUuids.map((sampleUuid) => (
+      [
+        { Key: `${projectUuid}/${sampleUuid}/barcodes.tsv.gz` },
+        { Key: `${projectUuid}/${sampleUuid}/features.tsv.gz` },
+        { Key: `${projectUuid}/${sampleUuid}/matrix.mtx.gz` },
+      ]
+    ));
```

Comment:
Ok, we will need to receive the samples object as a parameter in this function then because we cannot count on the dynamodb entries (currently they are updated before this function is called and placing this function before it would create an unnecessary and dangerous order dependency I think).